commit 728e0eaf99631d197e5158e21b4a8c4335a39231
Author: Tejun Heo <tj@kernel.org>
Date:   Fri Jul 2 14:41:24 2010 +0200

    ata_generic: drop hard coded DMA force logic for CENATEK
    
    Commit 1529c69adc (ata_generic: implement ATA_GEN_* flags and force
    enable DMA on MBP 7,1) implemented ATA_GEN_FORCE_DMA for forcing DMA
    mode and applied it to CENATEK but forgot to remove the original hard
    coded logic.  This is removal of redundant logic and doesn't affect
    correctness.
    
    Signed-off-by: Tejun Heo <tj@kernel.org>
    Reported-by: Andy Whitcroft <apw@canonical.com>
    Signed-off-by: Jeff Garzik <jgarzik@redhat.com>

diff --git a/drivers/ata/ata_generic.c b/drivers/ata/ata_generic.c
index 7107a69..cc5f772 100644
--- a/drivers/ata/ata_generic.c
+++ b/drivers/ata/ata_generic.c
@@ -54,7 +54,6 @@ static int generic_set_mode(struct ata_link *link, struct ata_device **unused)
 	const struct pci_device_id *id = ap->host->private_data;
 	int dma_enabled = 0;
 	struct ata_device *dev;
-	struct pci_dev *pdev = to_pci_dev(ap->host->dev);
 
 	if (id->driver_data & ATA_GEN_FORCE_DMA) {
 		dma_enabled = 0xff;
@@ -63,9 +62,6 @@ static int generic_set_mode(struct ata_link *link, struct ata_device **unused)
 		dma_enabled = ioread8(ap->ioaddr.bmdma_addr + ATA_DMA_STATUS);
 	}
 
-	if (pdev->vendor == PCI_VENDOR_ID_CENATEK)
-		dma_enabled = 0xFF;
-
 	ata_for_each_dev(dev, link, ENABLED) {
 		/* We don't really care */
 		dev->pio_mode = XFER_PIO_0;
